### PR TITLE
feature/workspace-server-side

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Tracking remaining work for v1.0-preview Workspace server side

Remaining work KPI
- [x] Add cancellation to fetch request
- [x] Add current filters to request
- [x] Rewrite configuration
- [ ] Error handling
- [x] Keep previous data

Remaining work Filter
- [x] Advanced filter
- [x] Initial loading state
- [x] Quick filter
- [x] Add filter item click
- [x] add clear button
- [x] add filter item label click
- [x] Quick filter search enter press
- [x] Hidden filters applied
- [x] Count (requires backend)
- [x] Keep previous data
- [x] Hidden filters applied
- [ ] Search (requires backend)
- [ ] Loading skeletons (UI)
- [ ] Error handling (UI)
- [ ] Custom value renders (UI)
- [ ] Default hidden filter in advanced (UI)
- [ ] Drag & drop advanced filter (UI)
- [x] Remove filteritems with count 0 (like in excel) (backend)
- [x] Filter bug atle, stale state
- [ ] Closed Quick filter doesnt show what is filtered.

Remaining work Garden
- [x] Expand column headers
- [x] Grouping
- [x] Subgrouping (max depth 2)
- [x] Highlight current week (frontend done) (backend not done)
- [x] Expand subgroup
- [x] 2D pagination
- [x] Highlight selected package
- [x] Rewrite configuration object
- [x] Comb all callback signatures for the garden or garden controller object
- [x] Add cancellation to all fetch requests
- [x] Loading skeletons
- [x] Add current filters to request
- [x] Refetch when filters change
- [x] Click -> Sidesheet
- [ ] Bookmarks
- [ ] Sorting
- [ ] Calculate column widths (backend) (ui)
- [ ] ItemWidth config

Remaining work Grid
- [x] Pagination
- [x] Infinite scroll
- [x] Add current filters to request
- [x] Refetch when filters change
- [x] Click -> Sidesheet
- [x] Rewrite configuration object
- [ ] Sorting
- [ ] Grouping
- [ ] Highlight selected
- [ ] Aggregation
- [ ] Bookmarks
- [ ] Error handling
- [x] export IServerSideGetRowsParams

Remaining work UX
- [ ] Approval

Extra
- [x] Remove context config on root
- [x] Expose filterstategroup
- [x] up the chunk size to 40 000 ish
- [x] Garden header group header wrong interface GardenHeaderGroup
- [x] Scrollbar way too long
- [ ] Pass grouping keys as prop to item view
- [x] Filter blinking
- [x] Filter stale state
- [ ] Presets

## Definition of Done
- [ ] PR title and description are to the point and understandable
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: packages/filter/src/lib/components/Filter.tsx [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
+  if (!context) {
+    console.error('Wrap the <Filter /> component in the <FilterContextProvider />');
     return null;
   }
-  return <QuickFilter controller={controller} />;
+
+  return (
+    <>
+      <Suspense
+        fallback={<div style={{ height: '48px', width: '100%', background: tokens.colors.ui.background__light.hex }} />}
```
Comment (Gustav-Eikaas) [inline packages/filter/src/lib/components/Filter.tsx:31]: add Component for this

File: packages/ag-grid/src/lib/components/Grid.tsx [hunk 2]
```diff
@@ -1,70 +1,33 @@
...
-
-  const rowData = useRowData(controller);
-
-  useSelectionService(controller, gridOptions.current?.api ?? undefined);
-  useColumnState(controller, gridOptions.current.columnApi ?? undefined);
-
   return (
     <StyledGridWrapper style={{ height }}>
       <AgGridReact
         rowHeight={32}
```
Comment (Gustav-Eikaas) [inline packages/ag-grid/src/lib/components/Grid.tsx:23]: Might have to go into gridOptions

File: packages/filter/src/lib/components/Filter.tsx [hunk 3]
```diff
@@ -1,17 +1,43 @@
...
+  const context = useContext(FilterContext);
+
+  if (!context) {
+    console.error('Wrap the <Filter /> component in the <FilterContextProvider />');
     return null;
   }
-  return <QuickFilter controller={controller} />;
+
+  return (
+    <ErrorBoundary FallbackComponent={() => <div>Filter failed to load</div>}>
```
Comment (jaysencpp) [inline packages/filter/src/lib/components/Filter.tsx:31]: TODO: Better fallback component

File: packages/filter/src/lib/components/filterQuickSearch/FilterQuickSearch.tsx [hunk 4]
```diff
@@ -9,59 +8,59 @@ export type SearchMode = 'id/desc' | 'all';
 export const FilterQuickSearch = (): JSX.Element => {
   const [searchText, setSearchText] = useState<string | undefined>();
 
-  const { clearSearch, search, valueFormatters } = useFilterContext();
+  // const searchOptions = [];
```
Comment (jaysencpp) [inline packages/filter/src/lib/components/filterQuickSearch/FilterQuickSearch.tsx:11]: TODO: ?

File: packages/filter/src/lib/components/quickFilter/QuickFilter.tsx [hunk 5]
```diff
@@ -9,54 +9,71 @@ import {
...
-  const calculateHiddenFiltersApplied = () =>
-    groups.reduce(
-      (acc, curr) => (!curr.isQuickFilter && getInactiveGroupValues(curr.name).length > 0 ? acc + 1 : acc),
-      0
-    );
+  const clearActiveFilters = () => {
+    setUncheckedValues([]);
+  };
+
+  const calculateHiddenFiltersApplied = () => 0;
```
Comment (jaysencpp) [inline packages/filter/src/lib/components/quickFilter/QuickFilter.tsx:72]: TODO

File: packages/filter/src/lib/components/quickFilter/QuickFilter.tsx [hunk 6]
```diff
@@ -66,16 +83,18 @@ export function QuickFilter<T>({ controller }: QuickFilterProps<T>): JSX.Element
             <FilterQuickSearch />
           </StyledLeftSection>
           <StyledRightSection>
-            {!isFilterExpanded && (
+            {true && (
```
Comment (jaysencpp) [inline packages/filter/src/lib/components/quickFilter/QuickFilter.tsx:86]: TODO

File: packages/garden/package.json [hunk 7]
```diff
@@ -20,6 +20,9 @@
     "@equinor/eds-core-react": "0.28.0",
     "@equinor/eds-icons": "^0.18.0",
     "@equinor/eds-tokens": "0.9.0",
+    "@tanstack/react-query": "^4.28.0",
+    "@tanstack/react-query-devtools": "^4.28.0",
```
Comment (jaysencpp) [inline packages/garden/package.json:24]: Dev tools should  be dev dep

File: packages/workspace-fusion/src/lib/integrations/status-bar/components/StatusBarWrapper.tsx [hunk 8]
```diff
@@ -1,26 +1,31 @@
...
+  TCustomSidesheetEvents extends BaseEvent = never,
+  TFilter = undefined
+>({ config }: StatusBarWrapperProps<TData, TContext, TCustomSidesheetEvents, TFilter>) {
+  const { filterState } = useFilterContext();
+
+  const { data, isLoading } = useQuery(['kpis', filterState], (a) => config(filterState, a.signal));
 
-  return <StatusBar items={config(data)} />;
+  if (isLoading || !data) {
+    return <div></div>;
```
Comment (jaysencpp) [inline packages/workspace-fusion/src/lib/integrations/status-bar/components/StatusBarWrapper.tsx:28]: Skeleton if loading?

File: packages/garden/package.json (hunk 7 above)
Comment (Gustav-Eikaas) [reply packages/garden/package.json:24]: todo remove

**Pull request closed without merging**